boards/arm/v2m_musca_s1/doc/index.rst

```diff
@@ -0,0 +1,401 @@
+
+- Implementation Defined Attribution Unit (`IDAU`_). The IDAU is used to define
+  secure and non-secure memory maps. By default, all of the memory space is
+  defined to be secure accessible only.
+- Secure and Non-secure peripherals via the Peripheral Protection Controller
+  (PPC). Peripherals can be assigned as secure or non-secure accessible.
+- Secure boot.
+- Secure `AMBA®`_ interconnect.
+
+Serial Configuration Controller (SCC)
+=====================================
+
+The ARM Musca-S1 test chip implements a Serial Configuration Control (SCC)
+register. The purpose of this register is to allow individual control of
+clocks, reset-signals and interrupts to peripherals, and pin-muxing.
+
+Boot memory
+================
+Normal Musca-S1 test chip boot operation is from 2MB eMRAM by default, and
+it offers the fastest boot method. Musca-S1 test chip also support to boot from
+32MB off-chip QSPI flash. You can update the DAPLink firmware and set the boot
+selector slider switch for either QSPI or eMRAM for booting.
+
+Programming and Debugging
+*************************
+
+Musca-S1 supports the v8m security extension, and by default boots to the
+secure state.
+
+When building a secure/non-secure application, the secure application will
+have to set the IDAU/SAU and MPC configuration to permit access from the
+non-secure application before jumping.
+
+The following system components are required to be properly configured during
+the secure firmware:
+
+- AHB5 TrustZone Memory Protection Controller (MPC).
+- AHB5 TrustZone Peripheral Protection Controller (PPC).
+- Implementation-Defined Attribution Unit (IDAU).
+
+For more details please refer to `Corelink SSE-200 Subsystem`_.
+
+Flashing
+========
+
+DAPLink
+---------
+
+V2M Musca-S1 provides:
+
+- A USB connection to the host computer, which exposes a Mass Storage and an
+  USB Serial Port.
+- A Serial Flash device, which implements the USB flash disk file storage.
+- A physical UART connection which is relayed over interface USB Serial port.
+
+This interfaces are exposed via DAPLink which provides:
+
+- Serial Wire Debug (SWD).
+- USB Mass Storage Device (USBMSD).
+- UART.
+- Remote reset.
+
+For more details please refer
+to the `DAPLink Website`_.
+
+
+Building a secure only application
+----------------------------------
+
+
+You can build applications in the usual way. Here is an example for
+the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: v2m_musca_s1
+   :goals: build
+
+Open a serial terminal (minicom, putty, etc.) with the following settings:
+
+- Speed: 115200
+- Data: 8 bits
+- Parity: None
+- Stop bits: 1
+
+Reset the board, and you should see the following message on the corresponding
+serial port:
+
+.. code-block:: console
+
+   Hello World! musca_s1
+
+Building a secure/non-secure with Trusted Firmware
+--------------------------------------------------
+
+The process requires five steps:
```

Comment:
Some mention of the `v2m_musca_s1_nonsecure` board option should probably be added here as the correct board name to use with TF-M.

Reply:
Done.